haystack/document_stores/search_engine.py

```diff
@@ -1142,6 +1144,154 @@ def _convert_es_hit_to_document(
+                                        "$and": {
+                                            "Type": "News Paper",
+                                            "Date": {
+                                                "$lt": "2019-01-01"
+                                            }
+                                        }
+                                    },
+                                    {
+                                        "$and": {
+                                            "Type": "Blog Post",
+                                            "Date": {
+                                                "$gte": "2019-01-01"
+                                            }
+                                        }
+                                    }
+                                ]
+                            }
+                            ```
+        :param top_k: How many documents to return
+        :param index: Index name for storing the docs and metadata
+        :param return_embedding: To return document embedding
+        :param headers: Custom HTTP headers to pass to elasticsearch client (e.g. {'Authorization': 'Basic YWRtaW46cm9vdA=='})
+                Check out https://www.elastic.co/guide/en/elasticsearch/reference/current/http-clients.html for more information.
+        :param scale_score: Whether to scale the similarity score to the unit interval (range of [0,1]).
+                            If true (default) similarity scores (e.g. cosine or dot_product) which naturally have a different value range will be scaled to a range of [0,1], where 1 means extremely relevant.
+                            Otherwise raw similarity scores (e.g. cosine or dot_product) will be used.
+        :return:
+        """
+        if index is None:
+            index = self.index
+
+        if return_embedding is None:
+            return_embedding = self.return_embedding
+
+        if headers is None:
+            headers = {}
+
+        if not self.embedding_field:
+            raise DocumentStoreError("Please set a valid `embedding_field` for OpenSearchDocumentStore")
+
+        if isinstance(filters, list):
+            if len(filters) != len(query_embs):
+                raise HaystackError(
+                    "Number of filters does not match number of query_embs. Please provide as many filters"
+                    " as query_embs or a single filter that will be applied to each query_emb."
+                )
+        else:
+            filters = [filters] * len(query_embs) if filters is not None else [{}] * len(query_embs)
+
+        body = []
+        for query_emb, cur_filters in zip(query_embs, filters):
+            cur_query_body = self._construct_dense_query_body(
+                query_emb=query_emb, filters=cur_filters, top_k=top_k, return_embedding=return_embedding
+            )
+            body.append(headers)
+            body.append(cur_query_body)
+
+        logger.debug("Retriever query: %s", body)
```

Comment:
What issue are we expecting to debug with this log? Do we need to add more details to debug this issue or this is enough?

Reply:
I just copied it from query_batch I guess. But OpenSearch's and elastic search's client already produce these logs. So you can achieve the same by just running something like
```python
logging.getLogger("opensearch").setLevel(logging.DEBUG)
```
I'll remove it.